src/parser/view_transform.c

```diff
@@ -4186,8 +4236,18 @@ mq_copypush_sargable_terms_helper (PARSER_CONTEXT * parser, PT_NODE * statement,
   is_outer_joined = mq_is_outer_join_spec (parser, spec);
 
   /* term(predicate) check */
+  in_spec = statement->info.query.q.select.from;
   for (term = statement->info.query.q.select.where; term; term = term->next)
     {
+      /* check for dblink's function term */
+      if (in_spec->info.spec.derived_table_type == PT_DERIVED_DBLINK_TABLE)
```

Comment:
Can't there be multiple tables in the From clause?
Wouldn't it be necessary to check in_spec->next (or_next?)?

Reply:
This routine is called for each dblinked spec. So, it's no need for inspection of all specs.